app/Http/Controllers/API/EventController.php

```diff
@@ -319,29 +354,416 @@ private function getUser() {
+     *     @OA\RequestBody(
+     *         @OA\MediaType(
+     *             mediaType="multipart/form-data",
+     *             @OA\Schema(
+     *                required={"start","end","title","description","location","lat","lng"},
+     *                @OA\Property(
+     *                   property="groupid",
+     *                   title="id",
+     *                   description="Unique identifier of the group to which the event belongs",
+     *                   format="int64",
+     *                   example=1
+     *                ),
+     *                @OA\Property(
+     *                   property="start",
+     *                   ref="#/components/schemas/Event/properties/start",
+     *                ),
+     *                @OA\Property(
+     *                   property="end",
+     *                   ref="#/components/schemas/Event/properties/start",
+     *                ),
+     *                @OA\Property(
+     *                   property="timezone",
+     *                   ref="#/components/schemas/Event/properties/timezone",
+     *                ),
+     *                @OA\Property(
+     *                   property="title",
+     *                   ref="#/components/schemas/Event/properties/title",
+     *                ),
+     *                @OA\Property(
+     *                   property="description",
+     *                   ref="#/components/schemas/Event/properties/description",
+     *                ),
+     *                @OA\Property(
+     *                   property="location",
+     *                   ref="#/components/schemas/Event/properties/location",
+     *                ),
+     *                @OA\Property(
+     *                   property="online",
+     *                   ref="#/components/schemas/Event/properties/online",
+     *                ),
+     *             )
+     *         )
+     *    ),
+     *    @OA\Response(
+     *        response=200,
+     *        description="Successful operation",
+     *        @OA\JsonContent(
+     *            @OA\Property(
+     *              property="data",
+     *              title="data",
+     *              ref="#/components/schemas/Event"
+     *            )
+     *        ),
+     *     )
+     *  )
+     */
+    public function createEventv2(Request $request)
+    {
+        $user = $this->getUser();
+
+        list($groupid, $start, $end, $title, $description, $location, $timezone, $latitude, $longitude, $online, $link) = $this->validateEventParams(
+            $request,
+            true
+        );
+
+        $group = Group::findOrFail($groupid);
+
+        // Check whether the event should be auto-approved, if all of the networks it belongs to
+        // allow it.
+        $autoapprove = $group->auto_approve;
+
+        if (!Fixometer::userCanCreateEvents($user))
+        {
+            // REVIEW: This doesn't check that they are a host of this particular group.  That is existing logic.
```

Comment:
Yes, it should here, I think.  Otherwise in theory it could be abused to create an event for a group you don't have rights to.

That said, I wouldn't **necessarily** change it in within userCanCreateEvents, as it's possible that that is a separate check to see whether a user can create events in general, not intended specifically for a given group.  I'm not sure, but we should check.

But ultimately I'd prefer to have a permission check here that is clearly named e.g. `userCanCreateEventsForGroup`

Reply:
I think this is part of the more general point that Restarters permissions are not well-defined and don't use idiomatic Laravel methods.  So I will change this to a TODO.